core/services/llo/retirement_report_cache.go

```diff
@@ -0,0 +1,151 @@
+package llo
+
+import (
+	"context"
+	"fmt"
+	sync "sync"
+
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+	ocr2types "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/sqlutil"
+)
+
+// RetirementReportCacheReader is used by the plugin-scoped
+// RetirementReportCache
+type RetirementReportCacheReader interface {
+	AttestedRetirementReport(cd ocr2types.ConfigDigest) ([]byte, bool)
+	Config(cd ocr2types.ConfigDigest) (Config, bool)
+}
+
+// RetirementReportCache is intended to be a global singleton that is wrapped
+// by a PluginScopedRetirementReportCache for a given plugin
+type RetirementReportCache interface {
+	services.Service
+	StoreAttestedRetirementReport(ctx context.Context, cd ocrtypes.ConfigDigest, seqNr uint64, retirementReport []byte, sigs []types.AttributedOnchainSignature) error
+	StoreConfig(ctx context.Context, cd ocr2types.ConfigDigest, signers [][]byte, f uint8) error
+	RetirementReportCacheReader
+}
+
+type retirementReportCache struct {
+	services.Service
+	eng *services.Engine
+
+	mu      sync.RWMutex
+	arrs    map[ocr2types.ConfigDigest][]byte
+	configs map[ocr2types.ConfigDigest]Config
+
+	orm RetirementReportCacheORM
+}
+
+func NewRetirementReportCache(lggr logger.Logger, ds sqlutil.DataSource) RetirementReportCache {
+	orm := &retirementReportCacheORM{ds: ds}
+	return newRetirementReportCache(lggr, orm)
+}
+
+func newRetirementReportCache(lggr logger.Logger, orm RetirementReportCacheORM) *retirementReportCache {
+	r := &retirementReportCache{
+		arrs:    make(map[ocr2types.ConfigDigest][]byte),
+		configs: make(map[ocr2types.ConfigDigest]Config),
+		orm:     orm,
+	}
+	r.Service, r.eng = services.Config{
+		Name:  "RetirementReportCache",
```

Comment:
Maybe this should include some kind of `LLO` prefix since it shows up as a root/global service in the hierarchy.